tron/bin/check_tron_jobs.py

```diff
@@ -293,41 +317,95 @@ def guess_realert_every(job):
     return realert_every
 
 
-def compute_check_result_for_job(client, job):
-    kwargs = {
-        "name": "check_tron_job.{}".format(job['name']),
-        "source": "tron",
+def sort_runs_by_interval(job_content, interval='day', until=None):
+    """ Sorts a job's runs by a time interval (day, hour, minute, or second),
+    according to a job run's run time.
+    """
+    interval_formats = {
+        'day': '%Y.%m.%d',
+        'hour': '%Y.%m.%d-%H',
+        'minute': '%Y.%m.%d-%H.%M',
+        'second': '%Y.%m.%d-%H.%M.%S',
     }
-    kwargs.update(job['monitoring'])
+
+    run_buckets = defaultdict(list)
+    if job_content is not None:
+        if not until:
+            until = time.time()  # can't set in default arg
+        if job_content['runs']:
+            earliest_run_time = min([
+                time.mktime(_timestamp_to_timeobj(run['run_time']))
+                for run in job_content['runs']
+            ])
+        else:
+            earliest_run_time = until
+
+        # We add all dates by interval between our earliest run_time and now,
+        # allowing functions downstream to see if some dates had no runs
+        start = datetime.datetime.fromtimestamp(earliest_run_time)
+        end = datetime.datetime.fromtimestamp(until)
+        step = datetime.timedelta(**{f'{interval}s': 1})
+        while start <= end:
+            run_buckets[start.strftime(interval_formats[interval])] = []
+            start += step
+
+        # Bucket runs by interval
+        for run in job_content['runs']:
+            # If interval is invalid, will raise a KeyError
+            run_time = time.strftime(
+                interval_formats[interval],
+                _timestamp_to_timeobj(run['run_time']),
+            )
+            run_buckets[run_time].append(run)
+    return dict(run_buckets)
+
+
+def compute_check_result_for_job(client, job):
+    kwargs = m(
+        name="check_tron_job.{}".format(job['name']),
+        source="tron",
+    )
     if 'realert_every' not in kwargs:
-        kwargs['realert_every'] = guess_realert_every(job)
-    kwargs['check_every'] = "{}s".format(_run_interval)
+        kwargs = kwargs.set('realert_every', guess_realert_every(job))
+    kwargs = kwargs.set('check_every', "{}s".format(_run_interval))
```

Comment:
Whoa, this is incorrect. We check every 5 minutes and this is "fixed"

Reply:
Given that the intent is that we don't want to allow users to set 'check_every' in any monitoring config files, I'll need to remove that attribute from `sensu_kwargs` on the next line of code.